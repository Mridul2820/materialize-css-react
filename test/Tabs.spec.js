import React from 'react';
import { render } from '@testing-library/react';
import Tabs from '../src/Tabs';
import Tab from '../src/Tab';
import mocker from './helper/new-mocker';

describe('Tabs', () => {
  const tabsInitMock = jest.fn();
  const tabInstanceDestroyMock = jest.fn();
  const tabsMock = {
    init: (el, options) => {
      tabsInitMock(options);
      return {
        destroy: tabInstanceDestroyMock
      };
    }
  };
  const restore = mocker('Tabs', tabsMock);

  const options = {
    duration: 400,
    onShow: jest.fn(),
    swipeable: true,
    responsiveThreshold: 2
  };

  afterAll(() => {
    restore();
  });

  beforeEach(() => {
    tabsInitMock.mockClear();
    tabInstanceDestroyMock.mockClear();
  });

  test('should create list of Tab items', () => {
    const { container } = render(
      <Tabs>
        <Tab title="one">One</Tab>
        <Tab title="Two">Two</Tab>
      </Tabs>
    );

    expect(container).toMatchSnapshot();
  });

  describe('with options', () => {
    test('initializes Tabs with options', () => {
      render(
        <Tabs options={options}>
          <Tab title="one">One</Tab>
          <Tab title="Two">Two</Tab>
        </Tabs>
      );

      expect(tabsInitMock).toHaveBeenCalledWith(options);
    });
  });

  describe('when unmounted', () => {
    test('should be destroyed', () => {
      const { unmount } = render(
        <Tabs options={options}>
          <Tab title="one">One</Tab>
          <Tab title="Two">Two</Tab>
        </Tabs>
      );
      unmount();

      expect(tabInstanceDestroyMock).toHaveBeenCalled();
    });
  });

  describe('children with idx', () => {
    test('consumes passed idx', () => {
      const { container } = render(
        <Tabs options={options}>
          <Tab idx="one" title="one">
            One
          </Tab>
          <Tab title="Two">Two</Tab>
        </Tabs>
      );

      expect(container.querySelector('#tab_one')).toBeTruthy();
    });
  });
});
