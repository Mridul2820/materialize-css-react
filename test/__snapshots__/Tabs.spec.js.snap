// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should create list of Tab items 1`] = `
<div>
  <ul
    class="tabs"
  >
    <li
      class="tab col"
    >
      <a
        class=""
        href="#tab_tabs-mockId0"
      >
        one
      </a>
    </li>
    <li
      class="tab col"
    >
      <a
        class=""
        href="#tab_tabs-mockId1"
      >
        Two
      </a>
    </li>
  </ul>
  <div
    class="row"
  >
    <div
      class="col s12"
      id="tab_tabs-mockId0"
    >
      One
    </div>
    <div
      class="col s12"
      id="tab_tabs-mockId1"
    >
      Two
    </div>
  </div>
</div>
`;
